### PR TITLE
Initialise Sumo Logic exporter

### PR DESCRIPTION
**Pull request by sumo-drosiek**
**Description:**
Initialise Sumo Logic exporter

**Link to tracking Issue:** #1498 

**Testing:**
Unit tests

**Documentation:**
Readme

### CODE REVIEW

File: exporter/sumologicexporter/config.go [hunk 1]
```diff
@@ -0,0 +1,130 @@
...
+
+// Config defines configuration for Sumo Logic exporter.
+type Config struct {
+	configmodels.ExporterSettings `mapstructure:",squash"` // squash ensures fields are correctly decoded in embedded struct.
+
+	// Global options
+	// Unique URL generated for your HTTP Source.
+	// This is the address to send data to.
+	// url: "https://collectors.sumologic.com/receiver/v1/http/<UniqueHTTPCollectorCode>"
+	URL string `mapstructure:"url"`
```
Comment (bogdandrutu) [inline exporter/sumologicexporter/config.go:32]: Please use `endpoint` as we use this term everywhere.

File: exporter/sumologicexporter/config.go [hunk 2]
```diff
@@ -0,0 +1,130 @@
...
+	// Option to enable compression (default true)
+	Compress bool `mapstructure:"compress"`
+	// Compression encoding format, either gzip or deflate (default gzip)
+	CompressEncoding CompressEncodingType `mapstructure:"compress_encoding"`
+	// Max HTTP request body size in bytes before compression (if applied).
+	// By default 1MB is recommended.
+	MaxRequestBodySize int `mapstructure:"max_request_body_size"`
```
Comment (bogdandrutu) [inline exporter/sumologicexporter/config.go:39]: If this is an http protocol, consider to use one the `confighttp` package for config (see https://github.com/open-telemetry/opentelemetry-collector/blob/master/exporter/otlphttpexporter/config.go) to have consistency and helper funcs.

File: exporter/sumologicexporter/config.go [hunk 3]
```diff
@@ -0,0 +1,130 @@
...
+	SourceName string `mapstructure:"source_name"`
+	// Desired host name.
+	// Useful if you want to override the source host configured for the source.
+	SourceHost string `mapstructure:"source_host"`
+	// Name of the client
+	Client string `mapstructure:"client"`
+
+	exporterhelper.TimeoutSettings `mapstructure:",squash"` // squash ensures fields are correctly decoded in embedded struct.
+	exporterhelper.QueueSettings   `mapstructure:"sending_queue"`
+	exporterhelper.RetrySettings   `mapstructure:"retry_on_failure"`
```
Comment (bogdandrutu) [inline exporter/sumologicexporter/config.go:69]: Consider to move these to the top close to the `ExporterSettings`. Keep kind of an order "standard embedded objects, custom objects"

File: exporter/sumologicexporter/config.go (hunk 1 above)
Comment (sumo-drosiek) [reply exporter/sumologicexporter/config.go:32]: Done

File: exporter/sumologicexporter/config.go (hunk 2 above)
Comment (sumo-drosiek) [reply exporter/sumologicexporter/config.go:39]: Done

File: exporter/sumologicexporter/config.go (hunk 3 above)
Comment (sumo-drosiek) [reply exporter/sumologicexporter/config.go:69]: Done

File: exporter/sumologicexporter/config.go (hunk 2 above)
Comment (sumo-drosiek) [reply exporter/sumologicexporter/config.go:39]: Done

File: exporter/sumologicexporter/config.go [hunk 4]
```diff
@@ -0,0 +1,125 @@
...
+type Config struct {
+	configmodels.ExporterSettings `mapstructure:",squash"` // squash ensures fields are correctly decoded in embedded struct.
+	confighttp.HTTPClientSettings `mapstructure:",squash"` // squash ensures fields are correctly decoded in embedded struct.
+	exporterhelper.QueueSettings  `mapstructure:"sending_queue"`
+	exporterhelper.RetrySettings  `mapstructure:"retry_on_failure"`
+
+	// Option to enable compression (default true)
+	Compress bool `mapstructure:"compress"`
+	// Compression encoding format, either gzip or deflate (default gzip)
+	CompressEncoding CompressEncodingType `mapstructure:"compress_encoding"`
```
Comment (bogdandrutu) [inline exporter/sumologicexporter/config.go:35]: I think you don't need both these options. You can have only CompressEncoding and:
- "" - disabled
- "gzip" - gzip
- "deflate" - deflate

**Review by bogdandrutu — APPROVED** [review]

**Comment by bogdandrutu** [comment]
@sumo-drosiek ping

File: exporter/sumologicexporter/config.go (hunk 4 above)
Comment (sumo-drosiek) [reply exporter/sumologicexporter/config.go]: Thats reasonable. It will simplify the further PR too, so I changed it.

**Comment by bogdandrutu** [comment]
@sumo-drosiek I think there is a problem with circle CI and this PR. Can you open a new Pr with the same changes?

**Comment by sumo-drosiek** [comment]
@bogdandrutu Sure, it can be due to using circle CI on our fork. I created PR from my personal account. It should works fine now.

Here is the PR #1565

**Pull request closed without merging**